src/components/pagination/index.js

```diff
@@ -0,0 +1,123 @@
+import BaseComponent from "../base-component.js";
+
+import "./pagination-style.css";
+
+export default class Pagination extends BaseComponent {
+  start = 0;
+  pageIndex = 0;
+  subElements = {};
+
+  constructor({ totalPages = 10, page = 1 } = {}) {
+    super();
+    this.totalPages = totalPages;
+    this.pageIndex = page - 1;
+    this.init();
+    this.addEventListeners();
+    this.update();
+  }
+
+  get template() {
+    return `
+      <nav class="os-pagination">
+        <a href="#" class="page-link previous" data-element="nav-prev">
+          <i class="bi bi-chevron-left"></i>
+        </a>
+
+        <ul class="page-list" data-element="pagination">
+
+        </ul>
+
+        <a href="#" class="page-link next" data-element="nav-next">
+          <i class="bi bi-chevron-right"></i>
+        </a>
+      </nav>
+    `;
+  }
+
+  goToPrevPage() {
+    if (this.pageIndex - 1 >= 0) {
+      this.dispatchEvent("page-changed", this.pageIndex - 1);
+    }
+  }
+
+  goToNextPage() {
+    if (this.pageIndex + 1 < this.totalPages) {
+      this.dispatchEvent("page-changed", this.pageIndex + 1);
+    }
+  }
+
+  addEventListeners() {
+    this.element.addEventListener("pointerdown", (event) => {
+      const navElement = event.target.closest('[data-element^="nav-"]');
+
+      if (navElement) {
+        const type = navElement.dataset.element;
+
+        if (type === "nav-prev") {
+          this.goToPrevPage();
+        }
+
+        if (type === "nav-next") {
+          this.goToNextPage();
+        }
+      }
+    });
+
+    this.element.addEventListener("pointerdown", (event) => {
+      const pageIndex = parseInt(event.target.dataset.pageIndex, 10);
+
+      if (!isNaN(pageIndex) && this.pageIndex !== pageIndex) {
+        this.pageIndex = pageIndex;
+
+        this.dispatchEvent("page-changed", pageIndex);
```

Comment:
**Security Issue**: The event handler directly modifies the pageIndex based on dataset input without validation, which can lead to Cross-Site Scripting (XSS) if the input is crafted maliciously. <br> **Fix**: Validate the 'pageIndex' extracted from the dataset to ensure it is a safe integer before using it to update the state and dispatch events. <br> **Code Suggestion**: 
 ```
 + const pageIndex = parseInt(event.target.dataset.pageIndex, 10);
 + if (!isNaN(pageIndex) && pageIndex >= 0 && pageIndex < this.totalPages) {
 +   this.pageIndex = pageIndex;
 +   this.dispatchEvent("page-changed", pageIndex);
 + }
 ```